### ISSUE TITLE
The model is not outputting sentences correctly

### ISSUE THREAD

**Issue by Jagyle** [michellesima/powerTransformer#1]
## ❓ Questions & Help

When I try to replicate the results of the study using generate.py after training the model, the out in the csv is not outputting coherent sentences. This is the case regardless of whether I train on 4 or 10 epochs or if I use the test, dev, or even part of the train set.

For the training I use: python train.py --setup joint

with the max_epochs changed to 10 when I'm training 10 epochs instead of 4.

For 10 epochs I use the command: python generate.py --dataset roc-test (or roc for the dev set) --epoch 10 --setup joint --vocabBoost

For 4 epochs I use the command: : python generate.py --dataset roc-test --epoch 4 --setup joint --vocabBoost

To be able to run the code I did have to add these two lines in train.py between lines 60 and 62:
avg = avg.cpu()
avg = avg.detach().numpy()

Also to be able to save the output from generate.py I had to change the path for savedfile from "gen_sen/joint-none-test.csv" to "./gen_sen/joint-none-test.csv"

[dev-set-10-epoch.csv](https://github.com/michellesima/powerTransformer/files/9905018/dev-set-10-epoch.csv)

[test-set-4-epoch.csv](https://github.com/michellesima/powerTransformer/files/9905014/test-set-4-epoch.csv)

[test-set-10-epoch.csv](https://github.com/michellesima/powerTransformer/files/9905021/test-set-10-epoch.csv)

[10-epoch-first-325-lines-in-train-dataset.csv](https://github.com/michellesima/powerTransformer/files/9905023/10-epoch-first-325-lines-in-train-dataset.csv)
